lib/semantics/expression.cc

```diff
@@ -1831,6 +1831,35 @@ MaybeExpr ExpressionAnalyzer::Analyze(const parser::Expr::DefinedBinary &) {
   return std::nullopt;
 }
 
+static void CheckFuncRefToArrayElementRefHasSubscripts(
+    semantics::SemanticsContext &context,
+    const parser::FunctionReference &funcRef) {
+  // Emit message if the function reference fix will end-up an array element
+  // reference with no subscripts because it will not be possible to later tell
+  // the difference in expressions between empty subscript list due to bad
+  // subscripts error recovery or because the user did not put any.
+  if (std::get<std::list<parser::ActualArgSpec>>(funcRef.v.t).empty()) {
+    auto &proc{std::get<parser::ProcedureDesignator>(funcRef.v.t)};
+    const auto *name{std::get_if<parser::Name>(&proc.u)};
+    name = {name
```

Comment:
I am surprised by these braces.  They may be affecting clang-format.  The statement might be more clear as an `if` (or with GNU's `?:` extension that we probably should not use).

Reply:
It was weird and clang-format made it even weirder indeed... I changed it to a simple `if`.